pkg/grizzly/grafana.go

```diff
@@ -21,17 +22,80 @@ type Folder struct {
 // Board enscapsulates a dashboard for upload to Grafana API
 type Board struct {
 	Dashboard map[string]interface{} `json:"dashboard"`
-	FolderID  int                    `json:"folderId"`
 	Overwrite bool                   `json:"overwrite"`
-	UID       string
-	Name      string
+}
+
+func (b *Board) UnmarshalJSON(data []byte) error {
+	var m map[string]interface{}
+	if err := json.Unmarshal(data, &m); err != nil {
+		return err
+	}
+
+	delete(m, "version")
+	delete(m, "id")
+
+	b.Dashboard = m
+	return nil
+}
+
+func (b Board) UID() string {
+	if s, ok := b.Dashboard["uid"]; ok {
+		return s.(string)
+	}
+	panic("no uid")
```

Comment:
this is probably a bit brief as a message isn't it?
```suggestion
	panic("No UID found. Grizzly requires a uid for all dashboards provided within the Jsonnet")
```

Reply:
It cannot happen if properly coded. This is only reached in case of a developer error.

Will make more verbose